contrib/coreos/user-data.example

```diff
@@ -176,3 +190,70 @@ write_files:
     content: |
       [Coredump]
       Storage=none
+  - path: /etc/systemd/system/ntpd.service.d/debug.conf
+    content: |
+      [Service]
+      ExecStart=
+      ExecStart=/usr/sbin/ntpd -g -n -f /var/lib/ntp/ntp.drift
+  - path: /opt/conf/fleetd-092-custom-binary.conf
+    content: |
+      [Service]
+      ExecStart=
+      ExecStart=/opt/bin/fleetd-0.9.2
+  - path: /opt/bin/graceful-shutdown.sh
+    permissions: '0755'
+    content: |
+      #!/usr/bin/bash
+      if [ "$1" != '--really' ]; then
+        echo "command must be run as: $0 --really"
+        exit 1
+      fi
+      # procedure requires the store-admin
+      ADMIN_RUNNING=$(docker inspect --format="{{ .State.Running }}" deis-store-admin)
+      if [ $? -eq 1 ] || [ "$ADMIN_RUNNING" == "false" ]; then
+        echo "deis-store-admin container is required for graceful shutdown"
+        exit 2
+      fi
+      set -e -x -o pipefail
+      # determine osd id
+      CURRENT_STATUS=$(/usr/bin/docker exec deis-store-admin ceph health | awk '{print $1}')
+      OSD_HOSTS=($(/usr/bin/etcdctl ls /deis/store/hosts/| awk -F'/' '{print $5}'))
+      for HOST in "${OSD_HOSTS[@]}"
+      do
+        PUBLIC_IP=$(/usr/bin/etcdctl get /_coreos.com/fleet/machines/`cat /etc/machine-id`/object | /opt/bin/jq -r '.PublicIP')
```

Comment:
Is this really the public IP? Can't we use `fleetctl list-machines` here, or the fleet HTTP API? https://github.com/coreos/fleet/blob/master/Documentation/api-v1.md#list-machines